### PR TITLE
Rspec work

### PR DESCRIPTION
**Pull request by Ngufor-emmanuella**
###In this mile stone of the project @Ngufor-emmanuella and @SaboorSohaib implemented the following;

1- Add unit tests for all the methods in all the classes:
- Book
- Person
- Student
- Teacher
- Classroom
- Rental

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  context 'When testing the Person' do
+    it 'create a new book when person class is initialised' do
+      person = Person.new(34, 'Duane David')
+      id = person.id
+      expect(person.age).to eq 34
+      expect(person.name).to eq 'Duane David'
+      expect(person.id).to eq id
+    end
+  end
+end
```
Comment (Felix45) [inline spec/person_spec.rb]: - [ Required ] Please consider adding unit tests for the following methods in the `Person` class. It is part of the project requirements that all methods in a class are tested.
  
  - `can_use_services?`
  - `correct_name`
  - `add_rental(book, date)`

## Project requirements
Screenshot from 2023-01-28 10-01-38

File: spec/students_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+describe Student do
+  context 'When testing the Student' do
+    it 'should create a new Student when student is initialised' do
+      student = Student.new(24, 'Willy Rovo', true)
+      expect(student.age).to eq 24
+      expect(student.name).to eq 'Willy Rovo'
+    end
+  end
+end
```
Comment (Felix45) [inline spec/students_spec.rb]: - [ Required ] Please consider adding unit tests for the following methods in the `Student` class. It is part of the project requirements that all methods in a class are tested.
  
  - `play_hooky`
  - `classroom`

File: spec/teacher_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      expect(teacher.age).to eq age
+    end
+
+    it 'should return the name of the teacher' do
+      expect(teacher.name).to eq name
+    end
+
+    it 'should return parent permission for teacher' do
+      expect(teacher.specialization).to eq specialization
+    end
```
Comment (Felix45) [inline spec/teacher_spec.rb]: - [ Required ] Please consider adding unit tests for the following methods in the `Teacher` class. It is part of the project requirements that all methods in a class are tested.
  
  - `can_use_services?`

File: spec/class_room.spec.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require_relative '../class_room'
+
+describe Classroom do
+  context 'testing the Classroom class' do
+    it 'create a new classroom when Classroom class is initialised' do
+      cl = Classroom.new('1E')
+      expect(cl.label).to eq '1E'
+    end
+  end
+end
```
Comment (Felix45) [inline spec/class_room.spec.rb]: - [ Required ] Please consider adding unit tests for the following methods in the `Classroom` class. It is part of the project requirements that all methods in a class are tested.
 
  - `add_student(student)`

File: spec/books_spec.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+describe Book do
+  context 'testing the Book class' do
+    it 'create a new book when book class is initialised' do
+      book = Book.new('title', 'author')
+      expect(book.title).to eq 'title'
+      expect(book.author).to eq 'author'
+    end
+  end
+end
```
Comment (Felix45) [inline spec/books_spec.rb]: - [ Required ] Please consider adding unit tests for the following methods in the `Book` class. It is part of the project requirements that all methods in a class are tested.
  
  - `add_rental(personal, date)`

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Ngufor-emmanuella  & @SaboorSohaib , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [ ] No linter errors :white_check_mark: 
- [ ] Professional readme file :white_check_mark: 
- [ ] Gitflow is used correctly :white_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @Ngufor-emmanuella 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [ ] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**